modules/create_person.rb

```diff
@@ -0,0 +1,50 @@
+require_relative '../student'
+require_relative '../teacher'
+require_relative '../app'
+
+module CreatePerson
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp.capitalize
+    print 'classroom: '
+    classroom = gets.chomp
+    print 'Has parent permission? (Y/N): '
+    parent_permission = gets.chomp.upcase
+    until %w[Y N].include?(parent_permission)
+      print 'Has parent permission? (Y/N): '
+      parent_permission = gets.chomp.upcase
+    end
+    student = Student.new(age, classroom, name, parent_permission: true) if parent_permission == 'Y'
+    student = Student.new(age, classroom, name, parent_permission: false) if parent_permission == 'N'
+    @people << student
+    puts 'Person(student) created successfully!'
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people << Teacher.new(age, specialization, name)
+    puts 'Person(Teacher) created successfully!'
+  end
+
+  def create_person
+    number = nil
+    until %w[1 2].include?(number)
+      print 'Do you want to create a student(1) or a teacher(2)? [Input the number 1 or 2 only]: '
+      number = gets.chomp
```

Comment:
Great job team 🤝

- [ ] As stated in the project requirements, could you ensure all requests for input and input validation should be placed in the `main. rb` file? This will make your methods respect the Single responsibility rule. Kindly implement this across all your methods, and ensure all user input and input validation is placed either in the `App. rb` file or the `Main. rb` file.
- Good examples of the Single Responsibility Rule.
![good](https://github.com/AbdusSattar-70/oop-school-library/assets/63917725/1ff07e2c-3d00-41fc-acc3-cc23c66747d5)

Reply:
Hello @AbdusSattar-70 ✋

- [ ] Kindly remove the code that accepts input from the user from your methods and put these prompts in the `main .rb` file. This will keep your methods smaller and they will only contain the logic. This way you will achieve the Single Responsibility Rule. Kindly take a look at the screenshots I left in the previous comment.

Reply:
@1GORDON Thanks for your valuable suggestion.